libcontainer/cgroups/systemd/common.go

```diff
@@ -365,22 +366,28 @@ func systemdVersion(conn *systemdDbus.Conn) (int, error) {
 			return
 		}
 
-		// verStr is like "v245.4-1.fc32" (including quotes)
-		if !strings.HasPrefix(verStr, `"v`) {
-			versionErr = fmt.Errorf("can't parse version %s", verStr)
-			return
-		}
-		// remove `"v` prefix and everything after the first dot
-		ver, err := strconv.Atoi(strings.SplitN(verStr[2:], ".", 2)[0])
-		if err != nil {
-			versionErr = err
-		}
-		version = ver
+		version, versionErr = systemdVersionAtoi(verStr)
+		return
 	})
 
 	return version, versionErr
 }
 
+func systemdVersionAtoi(verStr string) (int, error) {
+	// verStr should be of the form:
+	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32"
+	// all the input strings include quotes, and the output int should be 245
+	// thus, we unconditionally remove the `"v`
+	// and then match on the first integer we can grab
+	re := regexp.MustCompile(`"?v?([0-9]+)`)
```

Comment:
Strictly speaking, the quote is obligatory not optional, but I guess it does not matter much.